hasher-matcher-actioner/hmalib/common/classification_models.py

```diff
@@ -16,10 +16,21 @@ class Label:
     key: str
     value: str
 
+    def __eq__(self, other) -> bool:
+        # Concrete type doesnt matter for equality, only key, value
+        # eg the following are equal:
+        #   ClassificationLabel("Classification", "true_positive")
+        #   Label("Classification", "true_positive")
+        return (
+            isinstance(other, Label)
+            and self.key == other.key
+            and self.value == other.value
+        )
+
 
 @dataclass(unsafe_hash=True)
 class ClassificationLabel(Label):
-    key: str = field(default="Classification", init=False)
+    key: str
```

Comment:
As discussed, let's put things back the way they were a few PRs ago:

1. Put this back to having the defaulted field
2. Put the type back to Label for the sets in ActionRule
3. Change initializations of Classification objects to have a single parameter.
4. Anything else to get back to the previous state w.r.t. this topic.